File: TourPlanner.BL/TourLogService.cs
namespace TourPlanner.BL;

using TourPlanner.DAL;
using TourPlanner.Models;

public class TourLogService : ITourLogService
{
    private readonly ITourLogRepository _repository;

    public TourLogService(ITourLogRepository repository)
    {
        _repository = repository;
    }

    public IEnumerable<TourLog> GetByTourId(Guid tourId) => _repository.GetByTourId(tourId);
    public TourLog? GetById(Guid id) => _repository.GetById(id);
    public TourLog Add(TourLog log) => _repository.Add(log);
    public TourLog? Update(TourLog log) => _repository.Update(log);
    public bool Delete(Guid id) => _repository.Delete(id);
}

File: tourplanner-frontend/src/app/services/tour-log.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TourLog } from '../models/tour-log.model';

@Injectable({ providedIn: 'root' })
export class TourLogService {
  private readonly baseUrl = 'http://localhost:5210/api/tours';

  constructor(private http: HttpClient) {}

  getAll(tourId: string): Observable<TourLog[]> {
    return this.http.get<TourLog[]>(`${this.baseUrl}/${tourId}/logs`);
  }

  create(tourId: string, log: Partial<TourLog>): Observable<TourLog> {
    return this.http.post<TourLog>(`${this.baseUrl}/${tourId}/logs`, log);
  }

  update(tourId: string, logId: string, log: Partial<TourLog>): Observable<TourLog> {
    return this.http.put<TourLog>(`${this.baseUrl}/${tourId}/logs/${logId}`, log);
  }

  delete(tourId: string, logId: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${tourId}/logs/${logId}`);
  }
}
